### PR TITLE
:checkers spell documentation

### PR DESCRIPTION
**Pull request by ralacerda**

### CODE REVIEW

File: modules/checkers/spell/README.org [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#+TITLE:   checkers/spell
+#+DATE:    june 23, 2020
```
Comment (hlissner) [inline modules/checkers/spell/README.org:2]: Replace with `January 9, 2020`, which was the day this module was added/renamed from `:tools flyspell`.

File: modules/checkers/spell/README.org [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#+SINCE:   v2.0.9
+#+STARTUP: inlineimages nofold
+
+* Table of Contents :TOC_3:noexport:
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:7]: Please generate the ToC. It should happen on save with Doom's default org settings, but you can invoke it manually with `M-x toc-org-insert-toc`.

File: modules/checkers/spell/README.org [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+* Description
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:10]: Remove blank line immediately following heading

File: modules/checkers/spell/README.org [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
++ org
++ markdown
++ TeX
++ rst
++ mu4e-compose
++ message
++ git-commit
+
+** Maintainers
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:23]: Remove blank line immediately following heading

File: modules/checkers/spell/README.org [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+This module has no dedicated maintainers.
+
+** Module Flags
++ =+aspell= Use =aspell= as a backend for spellchecking.
++ =+hunspell= Use =hunspell= as a backend for spellchecking.
++ =+everywhere= Use spellcheck in every mode.
+
+** Plugins
+   # A list of linked plugins
```
Comment (hlissner) [inline modules/checkers/spell/README.org:32]: Remove comment

File: modules/checkers/spell/README.org [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
++ [[https://github.com/d12frosted/flyspell-correct][flyspell-corret]]
++ [[https://github.com/d12frosted/flyspell-correct#flyspell-correct-ivy-interface][flyspell-correct-ivy]] (=completion/ivy=)
++ [[https://github.com/d12frosted/flyspell-correct#flyspell-correct-helm-interface][flyspell-correct-helm]] (=completion/helm=)
++ [[https://github.com/d12frosted/flyspell-correct#flyspell-correct-popup-interface][flyspell-correct-popup]] (if *neither* =completion/ivy= or =completion/helm=)
++ [[https://github.com/rolandwalker/flyspell-lazy][flyspell-lazy]]
+
+* Prerequisites
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:40]: Remove blank line immediately following heading

File: modules/checkers/spell/README.org [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  This module requires either =aspell= or =hunspell= as backend.
+  It will automatically pick =aspell= if both are install.
+
+  You can specify the backend with the =+aspell= or =+hunspell= flag.
+
+* Features
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:47]: Remove blank line immediately following heading

File: modules/checkers/spell/README.org [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  + Ignores source code inside org documents.
+
+  When using =+everywhere=, =flyspell-prog-mode= will be automatically loaded for the following modes:
+  + yaml-mode-hook
+  + conf-mode-hook
+  + prog-mode-hook
+  =flyspell-prog-mode= will only spellcheck comments.
+
+* Configuration
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:60]: Remove blank line immediately following heading

File: modules/checkers/spell/README.org [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
++ [[https://github.com/d12frosted/flyspell-correct#flyspell-correct-helm-interface][flyspell-correct-helm]] (=completion/helm=)
++ [[https://github.com/d12frosted/flyspell-correct#flyspell-correct-popup-interface][flyspell-correct-popup]] (if *neither* =completion/ivy= or =completion/helm=)
++ [[https://github.com/rolandwalker/flyspell-lazy][flyspell-lazy]]
+
+* Prerequisites
+
+  This module requires either =aspell= or =hunspell= as backend.
+  It will automatically pick =aspell= if both are install.
+
+  You can specify the backend with the =+aspell= or =+hunspell= flag.
```
Comment (hlissner) [inline modules/checkers/spell/README.org:44]: Remove indentation

File: modules/checkers/spell/README.org [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  + Spellchecking and suggestions based on =aspell= or =hunspell=.
+  + Choosing suggestions using completion interfaces (=ivy= or =helm=).
+  + Lazily spellchecking recent changes only when idle.
+  + Ignores source code inside org documents.
+
+  When using =+everywhere=, =flyspell-prog-mode= will be automatically loaded for the following modes:
+  + yaml-mode-hook
+  + conf-mode-hook
+  + prog-mode-hook
+  =flyspell-prog-mode= will only spellcheck comments.
```
Comment (hlissner) [inline modules/checkers/spell/README.org:57]: Remove indentation

File: modules/checkers/spell/README.org [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+               markdown-mode-hook
+               git-commit-mode-hook)
+             #'flyspell-mode)
+  #+END_SRC
+
+#+BEGIN_SRC elisp
+  (remove-hook! '(markdown-mode-hook
+                  git-commit-mode-hook)
+             #'flyspell-mode)
+#+END_SRC
```
Comment (hlissner) [inline modules/checkers/spell/README.org:81]: Remove indentation

Comment (hlissner) [reply modules/checkers/spell/README.org:81]: And please reflow each paragraph with `M-x org-fill-paragraph`, so they're hard-wrapped.

File: modules/checkers/spell/README.org [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+
+
```
Comment (hlissner) [inline modules/checkers/spell/README.org:84]: Remove 2 of 3 blank newlines

File: modules/checkers/spell/README.org [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+* Troubleshooting
+  # Common issues and their solution, or places to look for help.
```
Comment (hlissner) [inline modules/checkers/spell/README.org:86]: Remove comment

File: modules/checkers/spell/README.org [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+  You can specify the backend with the =+aspell= or =+hunspell= flag.
+
+* Features
+
+  + Spellchecking and suggestions based on =aspell= or =hunspell=.
+  + Choosing suggestions using completion interfaces (=ivy= or =helm=).
+  + Lazily spellchecking recent changes only when idle.
+  + Ignores source code inside org documents.
+
+  When using =+everywhere=, =flyspell-prog-mode= will be automatically loaded for the following modes:
```
Comment (hlissner) [inline modules/checkers/spell/README.org:53]: Please reflow this paragraph with M-x org-fill-paragraph, so they're hard-wrapped.

File: modules/checkers/spell/README.org [hunk 15]
```diff
@@ -0,0 +1,86 @@
+#+TITLE:   checkers/spell
+#+DATE:    june 23, 2020
+#+SINCE:   v2.0.9
```
Comment (hlissner) [inline modules/checkers/spell/README.org:3]: Replace with v3.0.0

**Comment by hlissner** [comment]
Linking #1166 for posterity.

**Comment by ralacerda** [comment]
Fixed based on review

**Comment by hlissner** [comment]
Thanks!

**Pull request merged**